apps/src/templates/HourOfCodeGuideEmailDialog.jsx

```diff
@@ -0,0 +1,72 @@
+import React, {useState} from 'react';
+import PropTypes from 'prop-types';
+import {connect} from 'react-redux';
+import i18n from '@cdo/locale';
+import {SignInState} from '@cdo/apps/templates/currentUserRedux';
+import Button from '@cdo/apps/templates/Button';
+import AccessibleDialog from '@cdo/apps/templates/AccessibleDialog';
+import {Heading1, Heading3} from '@cdo/apps/componentLibrary/typography';
+import experiments from '@cdo/apps/util/experiments';
+import style from './accessible-dialogue.module.scss';
+
+function HourOfCodeGuideEmailDialog(signedIn) {
+  const [isOpen, setIsOpen] = useState(true);
+
+  const onClose = () => {
+    setIsOpen(false);
+  };
+
+  const bodyText = () =>
+    signedIn === true ? i18n.weHaveEverything() : i18n.signUpToReceiveGuide();
+
+  const emailGuideButtonText = () =>
+    signedIn === true ? i18n.emailMeAGuide() : i18n.getGuideContinue();
+
+  const continueWithoutEmailButtonText = () =>
+    signedIn === true ? i18n.continueToActivity() : i18n.continueWithoutGuide();
```

Comment:
Could `signedIn === true` be simplified to just `signedIn` or is there a worry of other types being passed in?

Reply:
It wasn't working that way (was acting as true if there was any value for signedIn) and I found examples of this pattern in our codebase so I figured if it works, it works!